src/sentry/sentry_apps/external_requests/issue_link_requester.py

```diff
@@ -60,60 +67,79 @@ class IssueLinkRequester:
-            raise SentryAppIntegratorError(
-                message=f"Unable to parse response from {self.sentry_app.slug}",
-                webhook_context={
-                    "error_type": "issue-link-requester.invalid-json",
-                    "uri": self.uri,
-                    "response": body,
-                    "installation_uuid": self.install.uuid,
-                },
-                status_code=500,
-            )
-        except RequestException as e:
-            error_type = "issue-link-requester.error"
-            extras = {
-                "sentry_app": self.sentry_app.slug,
-                "installation_uuid": self.install.uuid,
-                "project": self.group.project.slug,
-                "group": self.group.id,
+        event = SentryAppEventType(f"external_issue.{ACTION_TO_PAST_TENSE[self.action]}")
+        with SentryAppInteractionEvent(
+            operation_type=SentryAppInteractionType.EXTERNAL_REQUEST,
+            event_type=event,
+        ).capture() as lifecycle:
+            extras: dict[str, Any] = {
                 "uri": self.uri,
-                "error_message": str(e),
+                "installation_uuid": self.install.uuid,
+                "sentry_app_slug": self.sentry_app.slug,
+                "project_slug": self.group.project.slug,
+                "group_id": self.group.id,
             }
-            logger.info(error_type, extra=extras)
-            raise SentryAppIntegratorError(
-                message=f"Issue occured while trying to contact {self.sentry_app.slug} to link issue",
-                webhook_context={"error_type": error_type, **extras},
-                status_code=500,
-            )
-
-        if not self._validate_response(response):
-            raise SentryAppIntegratorError(
-                message=f"Invalid response format from sentry app {self.sentry_app} when linking issue",
-                webhook_context={
-                    "error_type": "issue-link-requester.invalid-response",
-                    "response": response,
-                    "installation_uuid": self.install.uuid,
-                    "uri": self.uri,
-                },
-                status_code=500,
-            )
-
-        return response
+            try:
+                request = send_and_save_sentry_app_request(
+                    self._build_url(),
+                    self.sentry_app,
+                    self.install.organization_id,
+                    event,
+                    headers=self._build_headers(),
+                    method="POST",
+                    data=self.body,
+                )
+                response_body = safe_urlread(request)
+
+                response = json.loads(response_body)
+            except (json.JSONDecodeError, TypeError) as e:
+                halt_reason = FAILURE_REASON_BASE.format(
+                    SentryAppExternalRequestHaltReason.BAD_RESPONSE
+                )
```

Comment:
nit: maybe make this into a variable and reuse instead of declaring this 3x?

Reply:
same thing with recording the halt 🤔

Reply:
okes to the reusing of the halt reason, though I think it's okay to keep the halt part since, imo, it's a bit easier to trace, that in this flow we recorded a halt for this use case.